bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -5596,10 +5611,16 @@ uint64_t RewriteInstance::getNewFunctionOrDataAddress(uint64_t OldAddress) {
       // If OldAddress is the another entry point of
       // the function, then BOLT could get the new address.
       if (BF->isMultiEntry()) {
-        for (const BinaryBasicBlock &BB : *BF)
-          if (BB.isEntryPoint() &&
-              (BF->getAddress() + BB.getOffset()) == OldAddress)
-            return BF->getOutputAddress() + BB.getOffset();
+        for (const BinaryBasicBlock &BB : *BF) {
+          if (BF->forEachEntryPoint(
+                  [&](uint64_t Offset, const MCSymbol *Symbol) {
+                    if (BB.isEntryPoint() &&
+                        (BF->getAddress() + BB.getOffset()) == OldAddress)
+                      return true;
+                    return false;
+                  }))
+            return BB.getOutputStartAddress();
+        }
```

Comment:
I meant to replace the outer loop with `forEachEntryPoint()`, but it doesn't really make the code much simpler. Please revert this change back to the original version. Sorry for confusion.

Reply:
No worries, I have now reverted this change.